src/com/cdd/bao/Main.java

```diff
@@ -235,7 +236,23 @@ private static void compileSchema(String[] options) throws Exception
 		Util.writeln("Loaded: " + schvoc.numTerms() + " terms.");
 		try (OutputStream ostr = new FileOutputStream(outputFile))
 		{
-			schvoc.serialise(ostr);
+			if (outputFile.endsWith(".json"))
+			{
+				JSONArray templates = new JSONArray();
+				for (Schema schema : schemata)
+				{
+					SchemaVocab subset = schvoc.singleTemplate(schema.getSchemaPrefix());
+				
+					JSONObject json = new JSONObject();
+					json.put("schemaPrefix", schema.getSchemaPrefix());
+					json.put("root", ClipboardSchema.composeGroup(schema.getRoot(), subset));
+					templates.put(json);
+				}
+				Writer wtr = new BufferedWriter(new OutputStreamWriter(ostr));
+				templates.write(wtr);
+				wtr.flush();
```

Comment:
Is there a reason to need the flush? the file is closed immediately afterwards.

Reply:
I think you're right, but the interplay between buffered things and normal streams isn't always seemless, and the difference between works vs. breaks isn't obvious. I usually don't like superfluous statements, but if the rules are unclear, then it may be the lesser evil.